src/librustc_mir/interpret/eval_context.rs

```diff
@@ -339,9 +345,24 @@ impl<'mir, 'tcx, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
         &self,
         value: T,
     ) -> T {
+        let mut param_env = self.param_env;
+        param_env.caller_bounds =
+            self.tcx.mk_predicates(param_env.caller_bounds.iter().filter(|predicate| {
+                match *predicate {
+                    ty::Predicate::Trait(..)
+                    | ty::Predicate::Subtype(..)
+                    | ty::Predicate::Projection(..)
+                    | ty::Predicate::WellFormed(..)
+                    | ty::Predicate::ObjectSafe(..)
+                    | ty::Predicate::ClosureKind(..)
+                    | ty::Predicate::ConstEvaluatable(..) => true,
+                    ty::Predicate::TypeOutlives(..) | ty::Predicate::RegionOutlives(..) => false,
+                }
```

Comment:
This was an initial hack to remove inference variables from the `param_env` before calling `subst_and_normalize_erasing_regions`. I since replaced that in a later commit with `infcx.at(..).normalize(..)`, which dies handle inference variables.